daisi/src/minhton-ns3/minhton_manager_scheduler_helper.cpp

```diff
@@ -458,6 +458,31 @@ void MinhtonManager::Scheduler::executeOneFailByPosition(uint16_t level, uint16_
   throw std::invalid_argument("position to fail to not found");
 }
 
+void MinhtonManager::Scheduler::executeOneFailByIndex(uint16_t index) {
+  std::cout << "\texecuteOneFailByIndex on index " << index << " at "
+            << Simulator::Now().GetMilliSeconds() << std::endl;
+
+  if (index >= manager_.node_container_.GetN()) {
+    throw std::invalid_argument("Index to fail out of range");
+  }
+
+  auto app =
+      manager_.node_container_.Get(index)->GetApplication(0)->GetObject<MinhtonApplication>();
+
+  if (app->getNodeInfo().isInitialized()) {
+    this->initiateFailureNow(index);
+    for (auto it = this->init_index_deque_.begin(); it != this->init_index_deque_.end(); it++) {
+      if (index == *it) {
+        this->init_index_deque_.erase(it);
+        break;
+      }
```

Comment:
To get the iterator at the correct position, this could be replaced with `std::find(..)` for better readability.

Reply:
Good point, this will be handled in a follow-up pull request together with other changes.